### PR TITLE
Add python /info endpoint

### PR DESCRIPTION
**Pull request by cbornet**

### CODE REVIEW

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py [hunk 1]
```diff
@@ -57,7 +65,153 @@ def fail_processing_on_permanent_errors(self):
...
+            self.send_header('Content-type', 'application/json')
+            self.end_headers()
+            self.wfile.write(json.dumps(self.agent_info.worker_status()).encode('utf-8'))
+        else:
+            self.send_response(404)
+            self.end_headers()
+
+
+def run_with_server(configuration, agent=None, agent_info: AgentInfo = AgentInfo(), max_loops=-1):
+    httpd = HTTPServer(('', 8081), partial(HttpHandler, agent_info))
```
Comment (cbornet) [inline langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py:203]: Is port 8081 OK ?

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py [hunk 2]
```diff
@@ -85,19 +240,28 @@ def run(configuration, agent=None, max_loops=-1):
...
+            'topic-source')
+
+    agent_info.source = source
 
     if hasattr(agent, 'write'):
-        sink = agent
+        sink = RuntimeSink(agent, agent_id, agent_type)
     else:
-        sink = TopicProducerSink(producer)
+        sink = RuntimeSink(TopicProducerSink(producer), 'topic-sink', 'topic-sink')
```
Comment (cbornet) [inline langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py:255]: It's the same agent-id/agent-type for Java topic producer.
Shoould we use different names ?

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py [hunk 3]
```diff
@@ -85,19 +240,28 @@ def run(configuration, agent=None, max_loops=-1):
...
+
+    agent_info.sink = sink
 
     if hasattr(agent, 'process'):
-        processor = agent
+        processor = RuntimeProcessor(agent, agent_id, agent_type)
     else:
-        processor = NoopProcessor()
+        processor = RuntimeProcessor(NoopProcessor(), "identity", "identity")
```
Comment (cbornet) [inline langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py:262]: It's the same agent-id/agent-type for Java identity processor.
Shoould we use different names ?

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/tests/test_runtime.py [hunk 4]
```diff
@@ -24,25 +25,97 @@
...
-    assert agent.context['close'] == 1
-    assert agent.context['set_commit_callback'] == 1
-    assert agent.context['records'][0].value() == 'some record 0 processed'
-    assert agent.context['records'][1].value() == 'some record 1 processed'
+    agent_info = runtime.AgentInfo()
+    runtime.run(config, agent=agent, agent_info=agent_info, max_loops=2)
+    print(json.dumps(agent_info.worker_status(), indent=2))
+    assert json.dumps(agent_info.worker_status(), indent=2) == """[
+  {
+    "agent-id": "testApplicationId-testAgentId",
```
Comment (cbornet) [inline langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/tests/test_runtime.py:47]: In Java the agent-id name here is testAgentId.
Should we change in Java or Python ?

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py [hunk 5]
```diff
@@ -85,19 +240,28 @@ def run(configuration, agent=None, max_loops=-1):
...
 
     if hasattr(agent, 'read'):
-        source = agent
+        source = RuntimeSource(agent, agent_id, agent_type)
     else:
-        source = TopicConsumerWithDLQSource(consumer, dlq_producer if dlq_producer else NoopTopicProducer())
+        source = RuntimeSource(
+            TopicConsumerWithDLQSource(consumer, dlq_producer if dlq_producer else NoopTopicProducer()),
+            'topic-source',
+            'topic-source')
```
[the remark this answers is not in the document]
Comment (eolivelli) [reply langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py:248]: it's perfect, the UI may expect those names

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py (hunk 2 above)
Comment (eolivelli) [reply langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py:255]: it's perfect, the UI may expect those names

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py (hunk 3 above)
Comment (eolivelli) [reply langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py:262]: it's perfect, the UI may expect those names

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/tests/test_runtime.py (hunk 4 above)
Comment (eolivelli) [reply langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/tests/test_runtime.py:47]: please change it in Python, the "agentId" should be unique in the scope of the application, not globally unique in the tenant

side topic: currently we have a problem if you explicitly set the same agentId in two pipelines of the same application, but the problem is to be solved in the planner

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py (hunk 1 above)
Comment (eolivelli) [reply langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py:203]: I think so

**Review by eolivelli — APPROVED** [review]
very good

+1 as far as we change the agentId

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (cbornet) [reply langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/runtime.py:248]: Yes but should they be different betweem Java and Python ?

File: langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/tests/test_runtime.py (hunk 4 above)
Comment (cbornet) [reply langstream-runtime/langstream-runtime-impl/src/main/python/langstream_runtime/tests/test_runtime.py]: done

**Pull request merged**